### ISSUE TITLE
Apply and enforce clang format

### ISSUE THREAD

**Issue by NoraHagmeyer** [imcs-compsim/MIRCO#20]
## Description
So far, no consistent layout is used. This might sometimes lead to pure layout changes of the code instead of functional changes. This makes MRs harder to review.

## Possible Solution and Definition of Done
- [x] Write and add a clang format files
- [x] Add specification on which version of clang format to use, i.e. via a python virtual environment
- [x] apply the clang format to the existing code base
- [x] Enforce the right clang format, i.e. during testing
- [x] Add information on how to integrate the clang format into your IDE to the project (readme or wiki?)

## Related Issues and Merge Requests

* Blocks
* Is blocked by
* Follows #15 
* Precedes
* Related to
* Part of
* Composed of

**Comment by RShaw026** [comment]
@mayrmt 
https://embeddedartistry.com/blog/2017/11/02/a-strategy-for-enforcing-formatting-with-your-build-server/

I used and tested the strategy in the above link mentioned by @NoraHagmeyer. The "code check" part of the build fails when the formatting is incorrect. Before starting a merge request, should we update to using clang-format version 14.0.0, as already done by BACI? Currently, we are using version 6.0.1.

> Since it does not seem possible/easy to apply specialized clang format in VSCode, @mayrmt found a few python scripts that could be adapted to aid in the development process.
> 
> https://github.com/mongodb/mongo/blob/a751344c0b46fa7ea121fb74aa931797b783ef2b/buildscripts/clang_format.py
> 
> https://embeddedartistry.com/blog/2017/11/02/a-strategy-for-enforcing-formatting-with-your-build-server/

**Comment by mayrmt** [comment]
@RShaw026 Yes, updating to a more recent version of `clang-format` makes sense. Good idea!
